ui/packages/consul-ui/app/abilities/nspace.js

```diff
@@ -16,7 +16,7 @@ export default class NspaceAbility extends BaseAbility {
   }
 
   get canChoose() {
-    return this.canUse && this.nspaces.length > 0;
+    return this.canUse && (this.nspaces.length || []) > 0;
```

Comment:
This was another 'while I was there':

When I added the PartitionAbility I noticed that (whilst its never the case in our app) partition/nspaces here could be null, so I just guarded it for safety incase it ever is the case in our app.

Reply:
I fixed a little typo here, main comment still stands

```javascript
(this.nspaces.length || [])
// should be
(this.nspaces || []).length
```

similar to the equivalent partitions one below.